module/x/auction/keeper/abci_test.go

```diff
@@ -0,0 +1,293 @@
+				oldBidAmount = auction.HighestBid.BidAmount
+			}
+			expectedModDifference = uint64(amount) - oldBidAmount
+		}
+
+		actualUserDifference := preGravBalAcc.Amount.Sub(postGravBalAcc.Amount)
+		actualModDifference := postGravBalMod.Amount.Sub(preGravBalMod.Amount)
+		require.True(t, actualModDifference.IsPositive(), "Module balance increased")
+		require.True(t, actualUserDifference.IsPositive(), "User balance decreased")
+		if rebid {
+			fmt.Printf("Expecting mod increase of %v - %v = %v \n", amount, auction.HighestBid.BidAmount, expectedModDifference)
+			require.Equal(t, expectedModDifference, actualModDifference.Uint64(), "module balance did not increase by exactly the rebid amount")
+			require.True(t, actualUserDifference.Uint64() >= expectedUserDifference, "user balance did not decrease by rebid amount + fee")
+		} else {
+			require.Equal(t, expectedModDifference, actualModDifference.Uint64())
+			require.True(t, actualUserDifference.Uint64() >= expectedUserDifference, "user balance did not decrease by amount + fee")
+		}
+	}
+}
+
+// Verifies that the given `auction` was paid to `expWinner` and their `winningBid` has been paid to the community pool
+// Optionally verifies that the community pool has received the bid amounts if `verifyPoolGrav` is true
+func VerifyAuctionPayout(suite *KeeperTestSuite, expWinner sdk.AccAddress, auction types.Auction, winningBid uint64, verifyPoolGrav bool) {
+	ctx := suite.Ctx
+	t := suite.T()
+	auctionKeeper := suite.App.AuctionKeeper
+
+	shouldBeNil := auctionKeeper.GetAuctionById(ctx, auction.Id)
+	require.Nil(t, shouldBeNil, "Discovered auction lingering in store after period ended")
+
+	awardBalance := auctionKeeper.BankKeeper.GetBalance(ctx, expWinner, auction.Amount.Denom)
+	require.True(t, awardBalance.IsGTE(auction.Amount), "winner %v has insufficient award token balance", expWinner.String())
+
+	communityPool := auctionKeeper.DistKeeper.GetFeePoolCommunityCoins(ctx)
+	poolCoin := communityPool.AmountOf(auction.Amount.Denom).TruncateInt()
+	require.True(t, poolCoin.IsZero(), "Positive community pool balance of reward token after auction success")
+
+	if verifyPoolGrav {
+		poolGrav := communityPool.AmountOf(GravDenom).TruncateInt()
+		expGrav := sdk.NewIntFromUint64(winningBid)
+		require.True(t, poolGrav.GTE(expGrav), "community pool does not have the bidders tokens")
+	}
+}
+
+// Asserts that the module's balances exactly match `coins`, and that no other coins are held by the module
+func AssertModuleBalanceStrict(suite *KeeperTestSuite, coins sdk.Coins) {
+	bankKeeper := suite.App.AuctionKeeper.BankKeeper
+	balances := bankKeeper.GetAllBalances(suite.Ctx, ModuleAccount)
+
+	require.Equal(suite.T(), coins, balances, "module balance does not match coins")
+}
+
+// Asserts that the pool contains the exact same amount of each coin provided in `coins`
+// Makes no assertions about other coins that may exist in the pool
+func AssertPoolBalanceRelaxed(suite *KeeperTestSuite, coins sdk.Coins) {
```

Comment:
these are nice utility functions, very useful. 

A quick unit test that proves these work as expected could be nice though.

Reply:
Added